### PR TITLE
Huawei push notifications

### PR DESCRIPTION
**Pull request by ignaciosantise**
## Summary

* Implemented push notifications in Huawei Devices
* Deleted `enablePushNotifications.js` because its deprecated

### Additional info
* Tested in RN v0.45.1

### CODE REVIEW

File: generators/app/tasks/installDependencies.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ const DEPENDENCIES = [
   'redux-thunk',
   'reselect',
   'seamless-immutable',
-  'react-native-dotenv'
+  'react-native-dotenv',
+  'react-native-huawei-protected-apps'
```
Comment (sbalay) [inline generators/app/tasks/installDependencies.js]: we should only install this dependency if push notifications feature is enabled. check this out https://github.com/Wolox/wolmo-bootstrap-react-native/pull/15/files#diff-237b39754a118903c976a389eb22a016R50

File: generators/app/templates/src/config/Huawei.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import HuaweiProtectedApps from 'react-native-huawei-protected-apps';
```
Comment (mvbattan) [inline generators/app/templates/src/config/Huawei.js]: I'd rename this file to `HuaweiPushNotifications`

File: generators/app/tasks/appSetup/pushNotificationsFeatureFiles.js [hunk 3]
```diff
@@ -14,4 +14,9 @@ module.exports = function pushNotificationsFeatureFiles() {
     this.templatePath('src', 'config', 'PushNotifications.js'),
     this.destinationPath(this.projectName, 'src', 'config', 'PushNotifications.js')
   );
+  // src/config/Huawei.js
+  this.fs.copy(
+    this.templatePath('src', 'config', 'Huawei.js'),
+    this.destinationPath(this.projectName, 'src', 'config', 'Huawei.js')
```
Comment (mvbattan) [inline generators/app/tasks/appSetup/pushNotificationsFeatureFiles.js]: `HuaweiPushNotifications.js` if you rename that file

File: generators/app/tasks/appSetup/enablePushNotifications.js [hunk 4]
```diff
@@ -1,17 +0,0 @@
-module.exports = function enablePushNotifications() {
```
Comment (mvbattan) [inline generators/app/tasks/appSetup/enablePushNotifications.js]: Comment that this file is deprecated (and hence is being deleted) in this PR's change log please :)

**Pull request merged**